src/main/java/com/hms_networks/americas/sc/extensions/system/application/SCAppManagement.java

```diff
@@ -56,4 +58,98 @@ public static void disableAppAutoRestart() {
     Logger.LOG_DEBUG("Disabling next JVM run command.");
     RuntimeControl.configureNextRunCommand(jvmCommand);
   }
+
+  /**
+   * Checks if a valid WAN IP address has been assigned to the device.
+   *
+   * @return true if an initialized WAN IP address is found.
+   */
+  public static boolean deviceHasWanIP() {
+    boolean hasWanIp = false;
+    final String uninitializedWan = "0.0.0.0";
+
+    if (NetManager.getWanIp().equalsIgnoreCase(uninitializedWan)) {
+      hasWanIp = true;
+    }
+    return hasWanIp;
+  }
+
+  /**
+   * Waits for a number of minutes for a WAN IP to be establlshed if one does not already exist.
+   * After the given time has passed, returns false indicating no WAN IP is present.
+   *
+   * @param timeOutMinutes returns true if an initialized WAN IP is detected
+   */
+  public static boolean waitForWanIp(int timeOutMinutes) {
+
+    // There are 20 seconds in a wait period. 3 periods exist in a minute
+    int minutesToPeriodsConverter = 3;
+    int maxWaitPeriods = timeOutMinutes * minutesToPeriodsConverter;
+
+    boolean wanConnectionEstablished = false;
+    // Ensure there is a valid WAN IP address
+    for (int i = 0; i < maxWaitPeriods; i++) {
+      Logger.LOG_INFO("WAN IP is: " + NetManager.getWanIp());
+
+      if (deviceHasWanIP()) {
+        // If no connection is established by end of wait time
+        if (i == (maxWaitPeriods - 1)) {
+          Logger.LOG_CRITICAL(
+              "No WAN connection detected, please ensure there is a WAN connection before starting"
+                  + " the connector.");
+          wanConnectionEstablished = false;
+          i = maxWaitPeriods;
+        }
+        // Wait 20 seconds on WAN connection initialization
+        waitForWanHelper();
+
+      } else {
+        Logger.LOG_INFO("WAN connection found.");
+        wanConnectionEstablished = true;
+        i = maxWaitPeriods;
+      }
+    }
+
+    return wanConnectionEstablished;
+  }
+
+  /**
+   * Pause current thread indefinitely if an initialized WAN IP has not been established. When a WAN
+   * IP is detected, resume application.
+   */
+  public static void waitForWanIp() {
```

Comment:
Same comments as above about using the logging functionality within the library itself. 

In the library, in most cases, a successful result should simply be indicated by returning a boolean `true` or just getting no error. If there is an error, the library should throw an exception.

This architecture allows any application which uses the library to decide how it handles errors in the manner it sees fit, which provides flexibility overall.

Reply:
After writing this comment, I realized that the methods directly above your additions also contain Logger.* calls. It's not relevant to this PR, but I would also like to see those Logger calls removed in the future. Architecturally, I really think it is up to the application itself to be handling logging.